### PR TITLE
fixed 

### LINKED ISSUE #1015: Modal doesn't close on the first click on the overlay after click on elements containing event.stopPropagation() inside modal
### Summary:
When opening a second modal from within a first modal, closing the second modal results in the first modal not closing on the first click on the overlay. Instead, it requires a second click to close. This issue is similar to issue #958. However, the solution provided there is not really suitable for our cross-platform project, which uses components from react-native, and I would like to avoid creating separate button components for the web modal, in which event.stopPropagation() would be removed, in order to achieve the correct behavior.

### Steps to reproduce:
1. Open the provided code sandbox: https://codesandbox.io/s/react-modal-layout-issue-xgfsvk
2. Click on the "Open Modal" button to open the first modal.
3. Click on the icon in the top left corner of the first modal to open the second modal.
4. Close the second modal by clicking on the overlay or close button.
5. Attempt to close the first modal by clicking on the overlay.

### Expected behavior:
The first modal should close after the first click on the overlay.

### Link to example of issue:
https://codesandbox.io/s/react-modal-layout-issue-xgfsvk

### Additional notes:
This issue also occurs when closing a modal by clicking on the close button, then reopening it and attempting to close it by clicking on the overlay. The modal only closes after the second click, not the first. This is a kind of standard behavior, so you can reproduce the problem for both the first and second modals. Below are the steps to reproduce the problem for the first modal.

### Steps to reproduce:
1. Open the provided code sandbox: https://codesandbox.io/s/react-modal-layout-issue-xgfsvk
2. Click on the "Open Modal" button to open the first modal.
3. Click on the icon in the top right corner of the first modal to close it.
4. Click on the "Open Modal" button again to open the first modal.
5. Attempt to close the first modal by clicking on the overlay.

### PR DESCRIPTION
**Pull request by msubham193**
Fixes #1015.

Changes proposed:

 added this.shouldClose = null; to the handleOverlayOnMouseDown method (src/components/ModalPortal.js:321) this  allow the modal to close in cases where the mouse press started on the overlay. Then, added the event.stopPropagation(); handleContentOnMouseDown method (src/components/ModalPortal.js:331)

Upgrade Path (for changed or removed APIs):

Acceptance Checklist:
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.
- [x] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

**Comment by ErikDeviashin** [comment]
Hi @msubham193!

Please read [contributing guidelines](https://github.com/reactjs/react-modal/blob/1a8f562437dfc5ad714080c891dd816eff6ba714/CONTRIBUTING.md) and rewrite PR subject and description accordingly. Rewrite the description wisely, describing what problem you solving, not how - it  can be seen in files changes anyway.

**Comment by ErikDeviashin** [comment]
Also will be good to fill Reviewers, Assignees and, if there is some labels in repo, Labels fields.

File: package-lock.json
Comment (ErikDeviashin) [inline package-lock.json:1]: This file shouldn't appear in this PR. With it you change many locked versions of lidraries, which affects many dependencies throughout project and can lead to version incompatibilities, unpredictable behavior, and/or deployment issues. Changes of this kind should only be made with the approval of the maintainers and when you really know what and why you are changing. You can read little more about package-lock.json file on [npm docs](https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json) site. Please remove it from the PR.

**Comment by diasbruno** [comment]
@msubham193 as @ErikDeviashin pointed out, there are a few things that needs to be clean up,

> Also will be good to fill Reviewers, Assignees and, if there is some labels in repo, Labels fields.

No need for this. It's just a simple patch.

To update the commit message, use `git commit --amend -m "new message"`.
To remove the package-lock.json, use `git checkout -- package-lock.json && git commit --amend`

If you run this command in sequence, it will do the right thing (hopefully).

**Comment by diasbruno** [comment]
Let me know if you need anything, @msubham193.

**Comment by diasbruno** [comment]
@ErikDeviashin Can you please help providing a test case for this patch? 
I think this is going to be the hardest part of this patch.

**Comment by ErikDeviashin** [comment]
@diasbruno 
No, it wasn't that hard, to be honest. Especially considering the many other test case examples available. Can you provide me access to push in the repository so I can push the changes? Or how would you like it to be done?
Also, I found that two other test cases (about mouse down on content/overlay and mouse up on the opposite element) weren't really working, so I fixed them too.

**Comment by diasbruno** [comment]
As you are interested in this fix, @ErikDeviashin, you can take over...

**Comment by diasbruno** [comment]
@msubham193 Thanks so much for helping with this. We are going to move on due to some changes that this PR needs.

**Pull request closed without merging**